core/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CollectHandles.scala

```diff
@@ -66,7 +62,7 @@ case class CollectHandles(child: Expression,
 
   // `CollectHandles` is non-deterministic since it's result depends on the
   // actual order of input rows.
-  override def deterministic: Boolean = false
+  override lazy val deterministic: Boolean = false
```

Comment:
Why is this lazy?

Reply:
Here, the lazy val just follows the deterministic value in abstract class Expression.